lib/virt_feature_test_base.pm

```diff
@@ -134,19 +191,27 @@ sub analyzeResult {
         $self->{test_results}->{$uncheckpoint_failure_guest}->{$uncheckpoint_failure}->{status} = 'FAILED';
         $self->{test_results}->{$uncheckpoint_failure_guest}->{$uncheckpoint_failure}->{error}  = $uncheckpoint_failure_error;
     }
+
+    if ($status eq 'PASSED' and !defined $self->{test_results}) {
+        $self->{test_results}->{'ALL GUESTS'}->{'ALL TESTS'}->{status} = 'PASSED';
+        $self->{test_results}->{'ALL GUESTS'}->{'ALL TESTS'}->{error}  = 'NONE';
+    }
 }
 
 sub post_fail_hook {
     my ($self) = shift;
     $self->{"stop_run"} = time();
     assert_script_run("history -w /root/commands_history");
+    $self->stop_monitor_guest_console;
     #(caller(0))[3] can help pass calling subroutine name into called subroutine
     $self->junit_log_provision((caller(0))[3]) if get_var("VIRT_AUTOTEST");
-    virt_utils::upload_supportconfig_log;
+    virt_utils::collect_host_and_guest_logs;
     $self->upload_coredumps;
-    #(caller(0))[0] can help pass calling package name into called subroutine
-    virt_utils::upload_virt_logs("/var/log/libvirt", (caller(0))[0] . "-libvirt-logs");
-    $self->SUPER::post_fail_hook;
+    upload_logs("/var/log/clean_up_virt_logs.log");
+    upload_logs("/var/log/guest_console_monitor.log");
```

Comment:
Is it better to have a dedicated directory under /var/log/ (/var/log/[xxx]) for all the debug logs like clean_up_virt_logs.log and guest_console_monitor.log? With it, upload logs/remove logs/future adding new logs will not need to change code by handling this root directory only.

Reply:
Unless these logs are used altogether, otherwise multiple upload of the same folder will introduce duplicate. So at the last, you still need to upload needed logs one by one. If you want to do so, you need to care about whether another one is already uploaded, otherwise there is duplication. Looks fancy, but you end up doing the same thing. I opt to do not do that.

Reply:
Sorry, I do not agree. A good design should be able to well avoid the problems you mentioned.

Reply:
I will answer your two concerns altogether here, including https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10850#discussion_r481852400. I think I already expressed some of the idea, the correct thing to do here is to wrap up scripts serve the same purpose into a standalone function which can be called later freely by user. What this standalone function does inside is up to the function itself, so no further folder rules here to be applied to the downloaded script and script's output, because there is no point to do that. Hope you can understand. It serves no purpose and waster unworthy time.

Reply:
I do not think you get my  point in #10850 (comment), as well as this one. I will explain separately in each PR discussion thread.

For this one, this is a minor suggestion. Just let it go.

Reply:
If you are going to repeat the same kind of thing, I have to say I understand. There is no further grouping for any logs there, for these scripts and these warpped up functions. I am repeating this to save your time.